.github/actions/is-pr-approved/src/main.ts

```diff
@@ -0,0 +1,167 @@
+import * as core from '@actions/core'
+import {Context} from '@actions/github/lib/context'
+import {GitHub} from '@actions/github/lib/utils'
+import {actionInputs, transformIfSet} from 'github-actions-utils'
+
+interface GitHubAPIError {
+  status: number
+  request: any // eslint-disable-line @typescript-eslint/no-explicit-any
+  response: any // eslint-disable-line @typescript-eslint/no-explicit-any
+}
+
+export const DEFAULT_ALLOWED_ACTORS: readonly string[] = [
+  'greenkeeper[bot]',
+  'dependabot[bot]',
+  'mergify[bot]',
+  'github-actions[bot]'
+]
+
+export const UNSUPPORTED_EVENT_ERROR =
+  'This action only support `push`, `workflow_dispatch`, `pull_request` and `pull_request_target` events'
+
+export async function run(): Promise<void> {
+  // Use dynamic import to let `github-action-ts-run-api` library inject
+  // Github Context before using it.
+  // Static import will not work and result in empty Github Context
+  const github = await import('@actions/github')
+
+  try {
+    // Validate event source
+    const eventName = github.context.eventName
+    if (['push', 'workflow_dispatch'].includes(eventName)) {
+      return core.info(`Skipping validation due to priviledge event (${eventName})`)
+    } else if (!['pull_request', 'pull_request_target'].includes(eventName)) {
+      return core.setFailed(UNSUPPORTED_EVENT_ERROR)
+    }
```

Comment:
This action should only allow actors that have enough privilege to run specific jobs or actions.
There are some cases where implicitly we know that the actor has enough privilege (`push` or workflow_dispatch event).
However, there are some cases where we can't implicitly know if the actor has enough privilege.
For example, this action can be run with `pull_request_review_comment` and bypass checks.
This could lead to unexpected behavior in some cases.

Reply:
Perhaps, the scope of this action should be redefined.
The scope is either a generic workflow or a workflow specific for pull_request/pull_request_target.
The goal of this action is to authorise an actor to perform some actions within a workflow based on the GitHub context.
From a user perspective, we could assume that this action can be used in any workflow and triggered by any event.
In that case, we can support some events where we can safely assume that the actor is authorised and otherwise deny by failing.
In addition, we can add an input to skip the validation for an explicit list of events.
WDYT about the above?

P.S: is-actor-authorise or similar seems a better name in this context.

Reply:
AFAIK, there is only the `issue_comment` event that can be interesting to support.
In that case, the action could be used to check if the actor is authorized to trigger the workflow.
Otherwise, we could assume that the workflow was triggered by an authorized actor and skip the validation.

Reply:
> there is only the issue_comment event that can be interesting to support.

That particular event won't need to be supported atm (see the internal issue linked to https://github.com/elastic/apm-agent-nodejs/pull/3258 that explains a bit the reason)